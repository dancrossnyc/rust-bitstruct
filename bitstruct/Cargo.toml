[package]
name = "bitstruct"
version = "0.1.1"
authors = ["Glenn Griffin <ggriffiniii@gmail.com>", "Dan Cross <crossd@gmail.com>"]
edition = "2018"
description = "Better Bitfields"
license = "MIT"

[dependencies]
bitstruct_derive = { version = "0.1.0", path = "../bitstruct_derive" }

[dev-dependencies]
trybuild = "1.0.42"
proptest = "1.0.0"
