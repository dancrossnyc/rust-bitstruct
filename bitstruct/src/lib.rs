pub use bitstruct_derive::bitstruct;