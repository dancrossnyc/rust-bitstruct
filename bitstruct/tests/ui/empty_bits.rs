use bitstruct::bitstruct;

bitstruct! {
    struct Reg(u32) {
        field1: u8 = 19 .. 19;
    }
}

fn main() {}

