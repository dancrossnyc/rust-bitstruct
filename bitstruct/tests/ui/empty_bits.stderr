error: empty bit range specified
 --> $DIR/empty_bits.rs:5:30
  |
5 |         field1: u8 = 19 .. 19;
  |                              ^
